app/controllers/provider/admin/referrer_filters_controller.rb

```diff
@@ -42,6 +42,6 @@ def find_cinstance
   end
 
   def return_url
-    admin_service_application_path(@cinstance.service, @cinstance)
+    provider_admin_application_path(@cinstance)
```

Comment:
There's only 1 show page and I moved it under `p/admin/applications` since it's the "main" context so to speak. We could add a show page for each context but IMO since a particular application does not depend on context, I would leave it like this.